src/services/indexedCollection.ts

```diff
@@ -0,0 +1,99 @@
+import { createSequence } from "@/createSequence"
+
+type IndexedCollectionId = number
+type CollectionData<TData> = Map<IndexedCollectionId, TData>
+type CollectionIndex<TData, TKey extends keyof TData> = Map<TData[TKey], Set<IndexedCollectionId>>
+
+type IndexedCollection<TData, TKeys extends keyof TData> = {
+  addItem: (item: TData) => void
+  deleteItem: <TKey extends TKeys>(index: TKey, value: TData[TKey]) => void
+  findItem: <TKey extends TKeys>(index: TKey, value?: TData[TKey]) => TData[]
+  clear: () => void
+}
+
+const getId = createSequence()
```

Comment:
I don't think there's a reason for this to be outside of the collection itself. Each collection can have its own id sequence right?